index.js

```diff
@@ -26,12 +26,10 @@ module.exports = function(source) {
   var svgo = new Svgo(config);
   svgo.optimize(source)
   .then(function(result) {
-    if (result.error) {
-      callback(new Error(result.error));
-      return;
-    }
-
     callback(null, result.data);
     return;
+  })
+  .catch(error => {
+    callback(new Error(error));
```

Comment:
Let's check if it's already `instanceof Error` and only wrap if it's not.

Also I'm one of those people who think that a `catch` like this is an antipattern because we may end up calling callback twice. I think we should use second argument of `then` instead. [More on the topic.](https://gist.github.com/mjackson/fafef431758f45ba3b830ad9cbeb2328#gistcomment-1761542)